src/dos/dos_locale_data.cpp

```diff
@@ -0,0 +1,2102 @@
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along
+ *  with this program; if not, write to the Free Software Foundation, Inc.,
+ *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+#include "dos_locale.h"
+
+#include "checks.h"
+#include "string_utils.h"
+
+CHECK_NARROWING();
+
+using namespace LocaleData;
+
+// NOTE: Locale settings below were selected based on our knowledge and various
+// public sources. Since we are only a small group of volunteers, not even
+// the linguists, and there are about 200 recognized countries in the world,
+// mistakes and unfortunate choices could happen.
+// Sorry for that, we do not mean to offend or discriminate anyone!
+
+// ***************************************************************************
+// Data structure methods
+// ***************************************************************************
+
+std::string CountryInfoEntry::GetMsgName() const
+{
+	const std::string base_str = "COUNTRY_NAME_";
+	return base_str + country_code;
+}
+
+// ***************************************************************************
+// Country info - time/date format, currency, etc.
+// ***************************************************************************
+
+// clang-format off
+const std::map<uint16_t, DosCountry> LocaleData::CodeToCountryCorrectionMap = {
+        // Duplicates listed here are mentioned in Ralf Brown's Interrupt List
+        // and confirmed by us using different COUNTRY.SYS versions:
+
+        { 35,  DosCountry::Bulgaria },
+        { 88,  DosCountry::Taiwan   }, // also Paragon PTS DOS standard code
+        { 112, DosCountry::Belarus  }, // from Ralph Brown Interrupt List
+        { 384, DosCountry::Croatia  }, // most likely a mistake in MS-DOS 6.22
+};
+// clang-format on
+
+// clang-format off
+const std::map<DosCountry, CountryInfoEntry> LocaleData::CountryInfo = {
+	{ DosCountry::International, { "International (English)", "XXA", { // stateless
+		{ LocalePeriod::Modern, {
+			// C
+			DosDateFormat::DayMonthYear, LocaleSeparator::Period,
+			DosTimeFormat::Time24H,      LocaleSeparator::Colon,
+			{ "$" }, "USD", 2,
+			DosCurrencyFormat::SymbolAmount,
+			LocaleSeparator::Space,         // thousands separator
+			LocaleSeparator::Period,        // decimal separator
+		} }, { LocalePeriod::Historic, {
+			// MS-DOS 6.22; country 61
+			DosDateFormat::DayMonthYear, LocaleSeparator::Slash,
+			DosTimeFormat::Time24H,      LocaleSeparator::Colon,
+			{ "$" }, "USD", 2,
+			DosCurrencyFormat::SymbolAmount,
+			LocaleSeparator::Comma,         // thousands separator
+			LocaleSeparator::Period,        // decimal separator
+			LocaleSeparator::Comma          // list separator
```

Comment:
Ok, so you're explicitly setting the list separators per language here. I don't really get why you needed that function I commented earlier that derives some other list separator programmatically. Why not define that here as well, or use something that's already defined?

Reply:
Function is now removed. I am now using semicolons if there is no definition for the given country.